From 1bde2de094a8fa2fa4ce7d4f5278d40525f4850a Mon Sep 17 00:00:00 2001
From: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
Date: Sun, 13 Mar 2022 22:50:12 +0300
Subject: [PATCH 034/151] arm64: dts: renesas: r8a779f0-vc4: add SPI flash chip

Add definitions for SPI flash chip connected to RPC.
Use quad mode for now.

Also add partition table compatible with the current bootloader setup.

Signed-off-by: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
---
 arch/arm64/boot/dts/renesas/r8a779f0-vc4.dts | 84 ++++++++++++++++++++
 1 file changed, 84 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a779f0-vc4.dts b/arch/arm64/boot/dts/renesas/r8a779f0-vc4.dts
index 910ef6baab12..99512c032acb 100644
--- a/arch/arm64/boot/dts/renesas/r8a779f0-vc4.dts
+++ b/arch/arm64/boot/dts/renesas/r8a779f0-vc4.dts
@@ -140,6 +140,17 @@
 		function = "pcie";
 	};
 
+	rpc_pins: rpc {
+		qspi0 {
+			groups = "qspi0_data4", "qspi0_ctrl";
+			function = "qspi0";
+		};
+		qspi1 {
+			groups = "qspi1_data4", "qspi1_ctrl";
+			function = "qspi1";
+		};
+	};
+
 	scif0_pins: scif0 {
 		groups = "scif0_data", "scif0_ctrl";
 		function = "scif0";
@@ -194,6 +205,79 @@
 	clock-frequency = <100000000>;
 };
 
+&rpc {
+	status = "okay";
+
+	pinctrl-0 = <&rpc_pins>;
+	pinctrl-names = "default";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <133000000>;
+		spi-rx-bus-width = <4>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			bootparam@0 {
+				reg = <0x00000000 0x00040000>;
+				read-only;
+			};
+			icumxa_loader@40000 {
+				reg = <0x00040000 0x00200000>;
+				read-only;
+			};
+			cert_header@240000 {
+				reg = <0x00240000 0x00040000>;
+				read-only;
+			};
+			secure_fw@280000 {
+				reg = <0x00280000 0x00100000>;
+				read-only;
+			};
+			icumh_payload@280000 {
+				reg = <0x00380000 0x00100000>;
+				read-only;
+			};
+			secure_monitor@480000 {
+				reg = <0x00480000 0x00080000>;
+				read-only;
+			};
+			rtos@500000 {
+				reg = <0x00500000 0x00400000>;
+				read-only;
+			};
+			g4mh_payload@900000 {
+				reg = <0x00900000 0x00100000>;
+				read-only;
+			};
+			space1@a00000 {
+				reg = <0x00a00000 0x00300000>;
+			};
+			uboot-env@d00000 {
+				reg = <0x00d00000 0x00040000>;
+			};
+			space2@d40000 {
+				reg = <0x00d40000 0x000c0000>;
+			};
+			uboot@e00000 {
+				reg = <0x00e00000 0x00200000>;
+				read-only;
+			};
+			tee@1000000 {
+				reg = <0x01000000 0x00100000>;
+				read-only;
+			};
+			space3@1100000 {
+				reg = <0x01100000 0x02f00000>;
+			};
+		};
+	};
+};
+
 &rswitch {
 	status = "okay";
 
-- 
2.17.1

